docs/transpile/create-a-transpiler-plugin.ipynb

```diff
@@ -0,0 +1,311 @@
+    "Our plugin simply runs [VF2Layout](/api/qiskit/qiskit.transpiler.passes.VF2Layout) for a number of trials that depends on the requested optimization level.\n",
+    "\n",
+    "First, we create a subclass of [PassManagerStagePlugin](/api/qiskit/qiskit.transpiler.preset_passmanagers.plugin.PassManagerStagePlugin). There is one method we need to implement, called [`pass_manager`](/api/qiskit/qiskit.transpiler.preset_passmanagers.plugin.PassManagerStagePlugin#pass_manager). This method takes as input a [PassManagerConfig](/api/qiskit/qiskit.transpiler.PassManagerConfig) and returns the pass manager that we are defining. The PassManagerConfig object stores information about the target backend, such as its coupling map and basis gates."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "f90299d9-5026-424c-b528-6d0defdddb54",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from qiskit.transpiler import PassManager\n",
+    "from qiskit.transpiler.passes import VF2Layout\n",
+    "from qiskit.transpiler.passmanager_config import PassManagerConfig\n",
+    "from qiskit.transpiler.preset_passmanagers import common\n",
+    "from qiskit.transpiler.preset_passmanagers.plugin import PassManagerStagePlugin\n",
+    "\n",
+    "\n",
+    "class MyLayoutPlugin(PassManagerStagePlugin):\n",
+    "    def pass_manager(\n",
+    "        self,\n",
+    "        pass_manager_config: PassManagerConfig,\n",
+    "        optimization_level: int | None = None,\n",
+    "    ) -> PassManager:\n",
+    "        layout_pm = PassManager(\n",
+    "            [\n",
+    "                VF2Layout(\n",
+    "                    coupling_map=pass_manager_config.coupling_map,\n",
+    "                    properties=pass_manager_config.backend_properties,\n",
+    "                    max_trials=optimization_level * 10 + 1,\n",
+    "                    target=pass_manager_config.target,\n",
+    "                )\n",
+    "            ]\n",
+    "        )\n",
+    "        layout_pm += common.generate_embed_passmanager(pass_manager_config.coupling_map)\n",
+    "        return layout_pm"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d7666879-14bc-479c-a91b-56a800897073",
+   "metadata": {},
+   "source": [
+    "Now, we expose the plugin by adding an `entry-points` table in `pyproject.toml` under the appropriate namespace. See the [table of transpiler plugin stages](/api/qiskit/transpiler_plugins#stage-table) for the entry points and expectations for each stage.\n",
+    "To expose this plugin in a package called `my_qiskit_plugin`, we would add the following in `pyproject.toml`:\n",
+    "\n",
+    "```toml\n",
+    "[project.entry-points.\"qiskit.transpiler.layout\"]\n",
+    "\"vf2\" = \"my_qiskit_plugin.module.plugin:MyLayoutPlugin\"\n",
+    "```"
```

Comment:
I updated the text to refer the reader to the setuptools documentation in case they are not using `pyproject.toml`.

Reply:
Thanks! I still think it would be a better user experience to use the tabs as it avoids context-switching, plus working out the syntax can be awkward if you don't regularly package projects (e.g. working out which strings need to be quoted etc.).

Reply:
My bad, looks like they need a `label` attribute (although not sure what it's used for). I've edited the original comment.

Reply:
Done.